modules/fundamental/src/package/model/details/package_version.rs

```diff
@@ -12,20 +20,24 @@ pub struct PackageVersionDetails {
     pub head: PackageVersionHead,
     pub base: PackageHead,
     pub packages: Vec<QualifiedPackageHead>,
+    pub advisories: Vec<PackageVersionAdvisory>,
 }
 
 impl PackageVersionDetails {
     pub async fn from_entity(
+        package: Option<package::Model>,
         package_version: &package_version::Model,
         tx: &ConnectionOrTransaction<'_>,
     ) -> Result<Self, Error> {
-        let package = package_version
-            .find_related(package::Entity)
-            .one(tx)
-            .await?
-            .ok_or(Error::Data(
-                "underlying package missing for package-version".to_string(),
-            ))?;
+        let package = if let Some(package) = package {
+            package
```

Comment:
nit: `unwrap_or_else`?

Reply:
Alas, can't or-else with an async closure, to load-if-None, yeah?

Reply:
Gotcha. The `if let` seems less clear (to me) than the `match` in this particular case, but like I said, a nit.

```rust
        let package = match package {
            Some(p) => p,
            None => package_version
                .find_related(package::Entity)
                .one(tx)
                .await?
                .ok_or(Error::Data("underlying package missing".to_string()))?,
        };
```